### PR TITLE
Button to reset cross-signing and SSSS keys

### PR DESCRIPTION
**Pull request by foldleft**
relates to https://github.com/vector-im/riot-web/issues/12221

### CODE REVIEW

File: src/CrossSigningManager.js [hunk 1]
```diff
@@ -116,17 +116,20 @@ export const crossSigningCallbacks = {
...
  */
-export async function accessSecretStorage(func = async () => { }) {
+export async function accessSecretStorage(func = async () => { }, force = false) {
     const cli = MatrixClientPeg.get();
     secretStorageBeingAccessed = true;
-
     try {
-        if (!await cli.hasSecretStorageKey()) {
+        if (!await cli.hasSecretStorageKey() || force) {
+            console.warn(!force ? "!hasSecretStorageKey()" : "force");
```
Comment (turt2live) [inline src/CrossSigningManager.js]: debugging?

File: src/CrossSigningManager.js [hunk 2]
```diff
@@ -116,17 +116,20 @@ export const crossSigningCallbacks = {
...
             // This dialog calls bootstrap itself after guiding the user through
             // passphrase creation.
             const { finished } = Modal.createTrackedDialogAsync('Create Secret Storage dialog', '',
                 import("./async-components/views/dialogs/secretstorage/CreateSecretStorageDialog"),
-                null, null, /* priority = */ false, /* static = */ true,
+                {
+                    force,
```
Comment (turt2live) [inline src/CrossSigningManager.js]: mildly impressed if the linter is complaining about this.

File: src/components/views/dialogs/ConfirmDestroyCrossSigningDialog.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        this.props.onFinished(false);
+    };
+
+    render() {
+        const BaseDialog = sdk.getComponent('views.dialogs.BaseDialog');
+        const DialogButtons = sdk.getComponent('views.elements.DialogButtons');
+
+        return (
+            <BaseDialog className='mx_ConfirmDestroyCrossSigningDialog' hasCancel={true}
+                        onFinished={this.props.onFinished}
```
Comment (turt2live) [inline src/components/views/dialogs/ConfirmDestroyCrossSigningDialog.js]: we align props 4 spaces from the last indentation, not aligned to other props. If you're using WebStorm, just don't have props on the same line as the tag.

File: src/CrossSigningManager.js (hunk 1 above)
Comment (foldleft) [reply src/CrossSigningManager.js]: yup

File: src/components/views/settings/CrossSigningPanel.js [hunk 4]
```diff
@@ -142,6 +157,12 @@ export default class CrossSigningPanel extends React.PureComponent {
                     {_t("Bootstrap cross-signing and secret storage")}
                 </AccessibleButton>
             </div>;
+        } else {
+            bootstrapButton = <div className="mx_CrossSigningPanel_buttonRow">
+                <AccessibleButton kind="danger" onClick={this._destroySecureSecretStorage}>
+                    {_t("Reset Secret Store and Cross-signing")}
```
Comment (jryans) [inline src/components/views/settings/CrossSigningPanel.js]: Let's use `secret storage` for consistency. Also, let's match the casing above, so perhaps:

```suggestion
                    {_t("Reset cross-signing and secret storage")}
```

**Comment by foldleft** [comment]
yeet

**Pull request merged**